gcp_variant_transforms/libs/resource_estimator.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for estimating the resources that a vcf_to_bq will require.
+
+Currently, the resource estimator only estimates the disk usage that a Dataflow
+pipeline will take along with the `MergeVariants` step, since this can cause
+expensive pipeline failures late in the run.
+"""
+import logging
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.filesystems import FileSystems
+
+from gcp_variant_transforms.beam_io import vcfio
+
+
+# TODO(hanjohn): Add unit tests.
+
+def _convert_variant_snippets_to_bytesize(variant):
+  # type: (vcfio.Variant -> int)
+  return coders.registry.get_coder(vcfio.Variant).estimate_size(variant)
+
+
+class SnippetSizeInfo(object):
+  def __init__(self,
+               raw_snippet_size,  # type: int
+               encoded_snippet_size,  # type: int
+              ):
+    # type: (...) -> (None)
+    self.raw = raw_snippet_size
+    self.encoded = encoded_snippet_size
+
+
+class FileSizeInfo(object):
+  def __init__(self, raw_file_size, encoded_file_size=None):
+    # type: (int, int) -> (None)
+    self.raw = raw_file_size
```

Comment:
s/self.raw/self._raw_size

Reply:
renamed to self.raw_size for now; since this is consumed by FileSizeInfoSumFn, though, I think this should remain public, WDYT?